app/views/users/new.html.erb

```diff
@@ -0,0 +1,22 @@
+<%= form_for @user do |f| %>
+  <%= f.label :username %>
+  <%= f.text_field :username %>
+  <br>
+  <%= f.label :email %>
+  <%= f.text_field :email %>
+  <br>
+  <%= f.label :password %>
+  <%= f.password_field :password %>
+  <br>
+  <%= f.label :password_confirmation, 'Confirm password' %>
+  <%= f.password_field :password_confirmation %>
+  <br>
+  <%= f.submit %>
+<% end %>
+
+<% if @user.errors.any? %>
+  <% @user.errors.full_messages.each do |error| %>
+    <%= error %>
+    <br>
+  <% end %>
+<% end %>
```

Comment:
Generally, I prefer the errors to render before the form so I immediately see that something went wrong with my submission and I need to check my inputs. That's just one man's opinion.